Marlin/ultralcd.cpp

```diff
@@ -576,17 +576,20 @@ static void lcd_status_screen() {
   #if ENABLED(BABYSTEPPING)
 
     int babysteps_done = 0;
+    float distance_done = 0.0f;
 
     static void _lcd_babystep(const AxisEnum axis, const char* msg) {
       ENCODER_DIRECTION_NORMAL();
       if (encoderPosition) {
-        int distance = (int32_t)encoderPosition * BABYSTEP_MULTIPLICATOR;
+        int babystep_increment = (int32_t)encoderPosition * BABYSTEP_MULTIPLICATOR;
+        float distance_increment = (float) babystep_increment / (float) planner.axis_steps_per_mm[axis];
         encoderPosition = 0;
         lcdDrawUpdate = LCDVIEW_REDRAW_NOW;
-        thermalManager.babystep_axis(axis, distance);
-        babysteps_done += distance;
+        thermalManager.babystep_axis(axis, babystep_increment);
+        babysteps_done += babystep_increment;
+        distance_done += distance_increment;
       }
-      if (lcdDrawUpdate) lcd_implementation_drawedit(msg, itostr3sign(babysteps_done));
+      if (lcdDrawUpdate) lcd_implementation_drawedit(msg, ftostr43sign(distance_done));
       if (LCD_CLICKED) lcd_goto_previous_menu(true);
```

Comment:
I kept it in the latest update to get rid of some irritating rounding error from incrementing a float.